packages/studio/src/components/Viewport/ViewportMenu.tsx

```diff
@@ -0,0 +1,49 @@
+import { useCallback } from "react";
+import { VIEWPORTS, Viewport } from "./defaults";
+import useStudioStore from "../../store/useStudioStore";
+
+interface OptionProps extends Viewport {
+  closeMenu: () => void;
+}
+
+export default function ViewportMenu(props: {
+  closeMenu: () => void;
+}): JSX.Element {
+  return (
+    <div className="absolute z-20 rounded bg-white text-sm text-gray-700 shadow-lg flex flex-col items-start py-1">
+      {Object.values(VIEWPORTS).map((val) => {
+        return <Option key={val.name} closeMenu={props.closeMenu} {...val} />;
+      })}
+    </div>
+  );
+}
+
+function Option({ name, closeMenu, styles }: OptionProps) {
+  const [setViewport] = useStudioStore((store) => [
+    store.pagePreview.setViewport,
+  ]);
+  const viewportOption = name;
```

Comment:
nit: you can do the rename above like `function Option({ name: viewportOption, which prevents the extra variable. Below the `name` variable is used in the callback but `viewportOption` is used in the markup when we probably just want to use one variable name